src/psyclone/psyGen.py

```diff
@@ -3697,3 +3814,54 @@ def name(self):
 
     def apply(self):
         return None, None
+
+
+class IfBlock(Node):
+    '''
+    Class representing an if-block within a Schedule.
+    '''
+    def __init__(self, parent=None):
+        super(IfBlock, self).__init__(parent=parent)
+        self._condition = ""
+
+    def __str__(self):
+        return "If-block: "+self._condition
+
+    @property
+    def coloured_text(self):
+        '''
+        Return text containing the (coloured) name of this node type
+
+        :return: the name of this node type, possibly with control codes
+                 for colour
+        :rtype: string
+        '''
+        return colored("If", SCHEDULE_COLOUR_MAP["If"])
+
+    def view(self, indent=0):
+        ''' Print representation of this node to stdout '''
+        print(self.indent(indent) + self.coloured_text + "[" +
+              self._condition + "]")
+        for entity in self._children:
+            entity.view(indent=indent + 1)
+
+
+class IfClause(IfBlock):
+    '''
+    Represents a sub-clause of an If block - e.g. an "else if()".
+    '''
+    def __init__(self, parent=None):
+        super(IfClause, self).__init__(parent=parent)
+        self._clause_type = ""  # Whether this is an else, else-if etc.
+
+    @property
+    def coloured_text(self):
+        '''
+        Return text containing the (coloured) name of this node type
+
+        :return: the name of this node type, possibly with control codes
+                 for colour
+        :rtype: string
```

Comment:
str not string? and include as part of return?

Reply:
As before: yes and no.